handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java

```diff
@@ -461,7 +464,7 @@ public final long sslCtxPointer() {
     //            get access to an OpenSslSessionContext after this method was called to prevent the user from
     //            producing a segfault.
     final void destroy() {
-        synchronized (ReferenceCountedOpenSslContext.class) {
+        if (DESTROYED_UPDATER.compareAndSet(this, 0, 1)) {
```

Comment:
We should also be careful with this approach though ... I'm not sure if this will happen but we should also be careful of the following:

Thread1:
long ctx = this.ctx; // volatile read

Thread2:
long ctx = this.ctx; // volatile read
this.ctx = 0; // volatile write
free(ctx);

Thread1:
someTcNativeMethod(ctx); // the memory pointed to by ctx has been freed!

Even if we do another volatile read at the end here ... it is possible a thread will use `ctx` in JNI while another thread destroys the memory. So we may need to leverage atomic reference counting (if offered by OpenSSL) on the ctx object ... or use synchronized.